### PR TITLE
Add VirtualMachine Details section to Instancetypes tab

### PR DESCRIPTION
**Pull request by pcbailey**
## 📝 Description

This PR adds the VirtualMachine Details section to the Instancetypes tab. It does not include the SSH section. That will be added in a separate PR.

### Empty
Selection_213

### Filled in
Selection_212

### CODE REVIEW

**Comment by pcbailey** [comment]
@metalice @upalatucci @hstastna PTAL

File: src/views/catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/SelectInstanceTypeSection.tsx [hunk 1]
```diff
@@ -20,7 +20,11 @@ const SelectInstanceTypeSection: React.FC<SelectInstanceTypeSectionProps> = ({
   setSelectedInstanceType,
 }) => {
   const handleSelect = (category: InstanceTypeCategory, size: InstanceTypeSize) => {
-    setSelectedInstanceType({ category, size, name: `${seriesDetails[category].prefix}.${size}` });
+    setSelectedInstanceType({
+      category,
+      size,
+      name: `${categoryDetailsMap[category].prefix}.${size}`,
```
Comment (hstastna) [inline src/views/catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/SelectInstanceTypeSection.tsx:26]: Maybe I'd better use `${categoryDetailsMap[category]?.prefix}.${size}`, as it's already used like this in `getInstancetypeString`, or are we sure there's gonna always be some `prefix`? I am not sure :)

File: src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/components/DetailsRightGrid.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import React, { useState } from 'react';
+
+import { V1beta1DataSource } from '@kubevirt-ui/kubevirt-api/containerized-data-importer/models';
+import { V1alpha1PersistentVolumeClaim } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { t } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { getPVC } from '@kubevirt-utils/resources/template/hooks/useVmTemplateSource/utils';
+import { DescriptionList, GridItem } from '@patternfly/react-core';
+import VirtualMachineDescriptionItem from '@virtualmachines/details/tabs/details/components/VirtualMachineDescriptionItem/VirtualMachineDescriptionItem';
+
+import { humanizeBinaryBytes } from '../../../../../../../../utils/utils/humanize';
```
Comment (upalatucci) [inline src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/components/DetailsRightGrid.tsx:10]:
```suggestion
import { humanizeBinaryBytes } from '@kubevirt-utils/utils/humanize';
```

File: src/views/catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/utils/utils.ts [hunk 3]
```diff
@@ -173,3 +174,31 @@ export const seriesDetails: InstanceTypesDetails = {
...
+};
+
+export const getInstancetypeDetails = (
+  category: InstanceTypeCategory,
+  size: InstanceTypeSize,
+): InstanceTypeSizeDetails => {
+  if (!category) return null;
+
+  const categoryInstancetypes = categoryDetailsMap[category]?.instanceTypes;
+  return categoryInstancetypes?.find((instancetype) => instancetype.name === size);
```
Comment (hstastna) [inline src/views/catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/utils/utils.ts:195]: Maybe `instancetype?.name` ?

File: src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/components/DetailsRightGrid.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  const [pvcSource, setPVCSource] = useState<V1alpha1PersistentVolumeClaim>();
+
+  const getPVCSource = async () => {
+    if (!bootSource) return;
+
+    const pvcData = bootSource?.spec?.source?.pvc;
+    const pvc = await getPVC(pvcData?.name, pvcData?.namespace);
+    setPVCSource(pvc);
+  };
+  getPVCSource().catch((e) => console.log(e));
```
Comment (upalatucci) [inline src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/components/DetailsRightGrid.tsx:27]:
```suggestion
  getPVCSource().catch(console.log);
```

File: src/views/catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/SelectInstanceTypeSection.tsx (hunk 1 above)
Comment (hstastna) [inline src/views/catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/SelectInstanceTypeSection.tsx:26]: Maybe I'd better use `${categoryDetailsMap[category]?.prefix}.${size}`, as it's already used like this in `getInstancetypeString`, or are we sure there's gonna always be some `prefix`? I am not sure :)

File: src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/components/DetailsLeftGrid.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  const { t } = useKubevirtTranslation();
+  const { preferences } = useInstanceTypesAndPreferences();
+  const preferencesMap = useMemo(() => convertResourceArrayToMap(preferences), [preferences]);
+  const operatingSystem = getOSFromDefaultPreference(bootSource, preferencesMap);
+
+  const { category, size } = instancetype;
+  const instancetypeSize: InstanceTypeSizeDetails = getInstancetypeDetails(category, size);
+
+  return (
+    <GridItem span={3}>
```
Comment (upalatucci) [inline src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/components/DetailsLeftGrid.tsx:41]: Maybe the GridItem can be in the [VMDetailsBody.tsx](https://github.com/kubevirt-ui/kubevirt-plugin/pull/1027/files#diff-f32c1009121572a83f8303944d6e99532d7f84bf56071ad52cc3683113cc05e6) so if we want to change grid with something else, we have to change just one component

**Review by upalatucci — COMMENTED** [review]
Just small things :-) 
Not something that would block this pr for me

File: src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/VMDetailsSection.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+.instancetypes-vm-details-section {
+  margin-top: var(--pf-global--spacer--lg);
+}
```
Comment (hstastna) [inline src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/VMDetailsSection.scss:3]: I can see "No newline at end of file".
So maybe adding a new line to this file would be better :)

File: src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/VMDetailsSection.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  return (
+    <div className="instancetypes-vm-details-section">
+      <ExpandableSection
+        toggleText={t('Advanced')}
+        data-test-id="expandable-advanced-section"
+        onToggle={() => setAdvancedSectionOpen(!advancedSectionOpen)}
+        isExpanded={advancedSectionOpen}
+        isIndented
+      >
+        Placeholder for SSH section
```
Comment (hstastna) [inline src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/VMDetailsSection.tsx:39]: I am not sure about this. Aren't we missing translations here? 
`{t('Placeholder for SSH section')}` ?

I expect to see some other text in here later (maybe this is WIP), but I'd use our `t`  function, if that's gonna be visible for the user (looks like it is gonna) .

File: src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/VMDetailsBody.scss [hunk 8]
```diff
@@ -0,0 +1,3 @@
+.instancetypes-vm-details-body {
+  margin-top: var(--pf-global--spacer--lg);
+}
```
Comment (hstastna) [inline src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/VMDetailsBody.scss:3]: Same as in the other css file ;) 
+1 for using Patternfly variables!

**Review by hstastna — COMMENTED** [review]
I've left some small comments, PTAL, otherwise great work! 👍🏽

File: src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/components/DetailsRightGrid.tsx (hunk 2 above)
Comment (pcbailey) [reply src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/components/DetailsRightGrid.tsx:10]: I had it that way, but got build errors. I'll try again.

Comment (pcbailey) [reply src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/components/DetailsRightGrid.tsx:10]: "Module not found: Error: Can't resolve '@kubevirt-utils/utils/humanize' in '/home/phillip/Development/kubevirt/kubevirt-plugin/src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/components"

File: src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/VMDetailsSection.tsx (hunk 7 above)
Comment (pcbailey) [reply src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/VMDetailsSection.tsx:39]: This is just a placeholder. I'm going to open another PR later today to replace that with the actual SSH component.

File: src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/components/DetailsRightGrid.tsx (hunk 4 above)
Comment (pcbailey) [reply src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/components/DetailsRightGrid.tsx]: Done!

File: src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/components/DetailsLeftGrid.tsx (hunk 5 above)
Comment (pcbailey) [reply src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/components/DetailsLeftGrid.tsx]: Good idea! Done!

File: src/views/catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/SelectInstanceTypeSection.tsx (hunk 1 above)
Comment (pcbailey) [reply src/views/catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/SelectInstanceTypeSection.tsx]: Done!

File: src/views/catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/utils/utils.ts (hunk 3 above)
Comment (pcbailey) [reply src/views/catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/utils/utils.ts]: Done!

File: src/views/catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/SelectInstanceTypeSection.tsx (hunk 1 above)
Comment (pcbailey) [reply src/views/catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/SelectInstanceTypeSection.tsx]: Done!

File: src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/VMDetailsSection.scss (hunk 6 above)
Comment (pcbailey) [reply src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/VMDetailsSection.scss]: Done!

File: src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/VMDetailsBody.scss (hunk 8 above)
Comment (pcbailey) [reply src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/VMDetailsBody.scss]: Done!

**Pull request merged**